service/header/core_listener.go

```diff
@@ -38,93 +38,59 @@ func NewCoreListener(p2pSub *P2PSubscriber, fetcher *core.BlockFetcher, dag form
 
 // Start kicks off the CoreListener listener loop.
 func (cl *CoreListener) Start(ctx context.Context) error {
-	cl.ctx, cl.cancel = context.WithCancel(context.Background())
+	if cl.blockSub != nil {
+		return fmt.Errorf("core-listener: already started")
+	}
+
+	sub, err := cl.fetcher.SubscribeNewBlockEvent(ctx)
+	if err != nil {
+		return err
+	}
 
-	go cl.listen(ctx)
```

Comment:
The problem with passing the context from `Start` to a long-living thing like `listen` is that it will be closed after 15 seconds. I feel like we've already been through this.